api/v1beta1/flowcollector_webhook_test.go

```diff
@@ -29,7 +29,6 @@ func TestBeta1ConversionRoundtrip_Loki(t *testing.T) {
 					Enable:             true,
 					InsecureSkipVerify: true,
 				},
-				BatchSize: 1000,
```

Comment:
For `beta1` -> `beta2` -> `beta1` cycle, we loose `BatchSize` value since this field moved from `Loki` to `Processor` spec.

If we want to manage such, we'll need to manually override `Convert_v1beta1_FlowCollector_To_v1beta2_FlowCollector` and `Convert_v1beta2_FlowCollector_To_v1beta1_FlowCollector` to copy the fields. 

Is it worth the price ? @jotak @msherif1234

Reply:
I think it's necessary to do it, else we loose the ability to set this field, since the conversions always happen between the served version (like v1beta2) and the stored version (v1beta1) ?

Reply:
(and same for all the other fields that are moved)

Reply:
also I tested deploying the sample flowcollector doesn't work, I think it still has all the old fields

Reply:
Hm, that's annoying since it always runs the conversion webhook, you end with `debug` fields in the yaml all the time even if you directly `v1beta2` sample:
```
spec:
  agent:
    ebpf:
      ...
      debug: {}
  consolePlugin:
    ...
    debug:
      port: 9001
      register: true
  processor:
    ...
    debug:
      port: 2055
      lokiMaxRetries: 2
      lokiMaxBackoff: 5s
      conversationTerminatingTimeout: 5s
      conversationEndTimeout: 10s
      lokiStaticLabels:
        app: netobserv-flowcollector
      enableKubeProbes: true
      lokiMinBackoff: 1s
      healthPort: 8080
      dropUnusedFields: true
      conversationHeartbeatInterval: 30s
```

I would prefer to omit these when default.

Reply:
I found a way to load defaults from CRD and show custom values only when set.
That required a bit of refactoring and creating a new [crd helper](https://github.com/netobserv/network-observability-operator/pull/467/files#diff-73ccb26d626eeea081eea7a3e9a8dd2fe3553bf80a746a1762a0c98467eff7d3) but it works very well and the user still have autocompletion in the yaml.

For simplicity, I have created [getters in flowcollector helper](https://github.com/netobserv/network-observability-operator/pull/467/files#diff-05471c458d8215e64e74c8e9123420dc94a81bddf0560d1536365c1ac95d68f9) that sets defaults and load custom values in their related config so we can rely on these everywhere in the code and we don't need to check for nils.